### PR TITLE
Refactor InstructionListAdapter and limit roundabout degrees in ManeuverView

### LINKED ISSUE #800: InstructionList flashes intermittently during reroute
In certain scenarios, the list of instructions (accessible by clicking the top banner), will flash when a reroute occurs while navigating.  We do have a UI treatment to show a rerouting state, but showing / hiding the list is not part of this.  Some other view is triggering this:

ezgif com-video-to-gif

### LINKED ISSUE #1061: Route instructions (overview) not correct with one waypoint
Hey,

**Android API:** 7.1.1 (Google Nexus 6)

**Mapbox Navigation SDK version:** com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.15.0

### Steps to trigger behavior

```
NavigationRoute.builder(this)
        .accessToken(Mapbox.getAccessToken())
        .origin(Point.fromLngLat(10.699819, 53.836633))
        .destination(Point.fromLngLat(10.705201, 53.834687))
        .voiceUnits(DirectionsCriteria.METRIC)
        .language(Locale.GERMAN)
        .profile(DirectionsCriteria.PROFILE_WALKING)
        .addWaypoint(Point.fromLngLat(10.700795, 53.837458))
        .build()
        .getRoute(new Callback<DirectionsResponse>() {
            @Override
            public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
	            directionsRoute = response.body().routes().get(0);
	            //...
	        }
	    });

 
NavigationLauncherOptions options = NavigationLauncherOptions.builder()
            .directionsRoute(directionsRoute)
            .shouldSimulateRoute(true)
            .enableOffRouteDetection(true)
            .lightThemeResId(R.style.CUNavigationViewLight)
            .build();

NavigationLauncher.startNavigation(this, options);
```

### Expected behavior
The navigation instruction overview should show the overview from second route-part after reaching first waypoint to.

### Actual behavior
Banner and voice instructions are correct. But there is no refresh in the overview - the old instructions are still displayed. 
After reaching the waypoint at "Universitätsstraße" the next overview should contain something like "Marie-Curie-Straße"

ezgif com-video-to-gif-2

### PR DESCRIPTION
**Pull request by danesfeder**
Closes #800 Closes #1061 

If the `ManeuverView` is drawing a roundabout and a `roundaboutAngle` is not provided, it will draw something like:

screen shot 2018-06-26 at 11 07 28 am

This PR adds a default 90 degrees as well as a limit, to the roundabout icon is not drawn over 300 degrees - after 300 degrees, the arrow begins to overlap with the beginning of the shaft of the icon and doesn't look great.  This icon is drawn @ 300 degrees: 

screen shot 2018-06-26 at 11 05 08 am

### CODE REVIEW

**Comment by danesfeder** [comment]
@Cjballard in some scenarios, like showing instructions in the list dropdown, we don't have the exact degree information.  Does defaulting to 90 degrees seem like the best UI / UX treatment for this scenario?

**Comment by Guardiola31337** [comment]
> Does defaulting to 90 degrees seem like the best UI / UX treatment for this scenario?

IMO we should not. It could be confusing to users. What about drawing a blurry one indicating that we don’t have the data of the angle? Any other ideas to indicate that @Cjballard?

**Comment by Cjballard** [comment]
Hmm. That's tricky. We don't necessarily want to set users up for an expectation of where the exit will be via the icon. I've seen icons like this in the wild, which could indicate that there's a turn in the circle but not designating where, which could be a good option. We'd have to create a new icon for this one I think.

img_27d4d08bbec0-1


Do we get any degree information at all or just that it's an exit in the circle? Maybe not showing the entry point on the icon or having a generic traffic circle icon without a maneuver in the list instructions?

**Comment by Guardiola31337** [comment]
@Cjballard 
> Do we get any degree information at all or just that it's an exit in the circle?

Nope, we get nothing. The `degrees` field is `null` in those situations.

> Maybe not showing the entry point on the icon or having a generic traffic circle icon without a maneuver in the list instructions?

Yeah, I think this is a great idea!

cc @danesfeder

**Comment by danesfeder** [comment]
@Cjballard @Guardiola31337, I think we may be able to squash two bugs here.  If we populate the list with a list of `BannerInstructions`, we will have the degree data and, if we get the logic right, we can also squash https://github.com/mapbox/mapbox-navigation-android/issues/1061 (I've confirmed it seems our logic is getting out of sync).

Let's dig on this and then, if all else fails, look into an icon like Chris mentioned above.  Sound good?

**Comment by Guardiola31337** [comment]
> I think we may be able to squash two bugs here.  If we populate the list with a list of `BannerInstructions`, we will have the degree data and, if we get the logic right, we can also squash https://github.com/mapbox/mapbox-navigation-android/issues/1061 (I've confirmed it seems our logic is getting out of sync).

Good point!

> Let's dig on this and then, if all else fails, look into an icon like Chris mentioned above.

Sounds good to me 👍

**Comment by danesfeder** [comment]
The instruction list was flashing (even with visibility set to `GONE`) because we were constantly updating it, even if not visible.  I was able to address this by only updating the list when it's visible:

ezgif com-video-to-gif

🎉

**Comment by danesfeder** [comment]
@Guardiola31337 this is ready for 👁 now

File: libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/TestRouteBuilder.java [hunk 1]
```diff
@@ -1,17 +1,70 @@
...
     ClassLoader classLoader = getClass().getClassLoader();
     InputStream inputStream = classLoader.getResourceAsStream(filename);
     Scanner scanner = new Scanner(inputStream, UTF_8.name()).useDelimiter("\\A");
     return scanner.hasNext() ? scanner.next() : "";
   }
+
+  DirectionsRoute buildTestDirectionsRoute(@Nullable String fixtureName) throws IOException {
+    fixtureName = checkNullFixtureName(fixtureName);
+    Gson gson = new GsonBuilder()
+      .registerTypeAdapterFactory(DirectionsAdapterFactory.create()).create();
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/TestRouteBuilder.java]: NIT - unnecessary new line

**Comment by danesfeder** [comment]
@Guardiola31337 this is updated and ready for another round of 👀

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/summary/list/InstructionListPresenter.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import java.util.List;
+
+public class InstructionListPresenter {
+
+  private static final int TWO_LINES = 2;
+  private static final int ONE_LINE = 1;
+  private static final float TWO_LINE_BIAS = 0.65f;
+  private static final float ONE_LINE_BIAS = 0.5f;
+  private static final int FIRST_INSTRUCTION_INDEX = 0;
+
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/summary/list/InstructionListPresenter.java]: NIT - unnecessary empty line

**Comment by danesfeder** [comment]
@Guardiola31337 should be addressed now:

ezgif com-video-to-gif

**Pull request merged**